packages/analytics-js/src/components/core/Analytics.ts

```diff
@@ -188,8 +191,13 @@ class Analytics implements IAnalytics {
             this.onInitialized();
             break;
           case 'loaded':
-            this.loadDestinations();
             this.processBufferedEvents();
+            // Short-circuit the life cycle and move to the ready state if pre-consent behavior is enabled
+            if (state.consents.preConsent.value.enabled === true) {
+              state.lifecycle.status.value = 'ready';
+            } else {
+              this.loadDestinations();
```

Comment:
Yes, because, both the queues would have been initialized and can consume events. While the cloud mode queue might start processing events, the device mode queue is only started after destinations are initialized.